x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/analyzer/AnalyzerRules.java

```diff
@@ -39,7 +40,7 @@ protected LogicalPlan rule(Filter filter) {
         }
 
         private Expression replaceRawBoolFieldWithEquals(Expression e) {
-            if (e instanceof FieldAttribute) {
+            if (e instanceof FieldAttribute && e.dataType() == BOOLEAN) {
```

Comment:
The rule could check whether the `Filter` is resolved before applying the equality.

Reply:
Ah, right. Thanks, fixed.